pkg/egressservice/egressservice_test.go

```diff
@@ -0,0 +1,515 @@
+		},
+		{
+			name:                egressTestSvcName,
+			nsname:              "",
+			sourceIPBy:          "LoadBalancerIP",
+			addToRuntimeObjects: false,
+			client:              true,
+			expectedError:       fmt.Errorf("egressService's namespace cannot be empty"),
+		},
+		{
+			name:                egressTestSvcName,
+			nsname:              egressTestSvcNamespace,
+			sourceIPBy:          "LoadBalancerIP",
+			addToRuntimeObjects: false,
+			client:              false,
+			expectedError:       fmt.Errorf("egressService's 'apiClient' cannot be empty"),
+		},
+		{
+			name:                egressTestSvcName,
+			nsname:              egressTestSvcNamespace,
+			sourceIPBy:          "LoadBalancerIP",
+			addToRuntimeObjects: false,
+			client:              true,
+			expectedError: fmt.Errorf("egressService object %q does not exist in namespace %q",
+				egressTestSvcName, egressTestSvcNamespace),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testEgressService := buildDummyEgressService(testCase.name, testCase.nsname, testCase.sourceIPBy)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testEgressService)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: testSchemes,
+			})
+		}
+
+		testEgressServiceBuilder, err := Pull(testSettings, testCase.name, testCase.nsname)
+
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.name, testEgressServiceBuilder.Definition.Name)
+			assert.Equal(t, testCase.nsname, testEgressServiceBuilder.Definition.Namespace)
+			assert.Equal(t, testCase.sourceIPBy, string(testEgressServiceBuilder.Definition.Spec.SourceIPBy))
+		}
+	}
+}
+
+func TestGet(t *testing.T) {
+	testCases := []struct {
+		name                string
+		nsname              string
+		sourceIPBy          string
+		addToRuntimeObjects bool
+		client              bool
+		expectedError       error
+	}{
+		{
+			name:                egressTestSvcName,
+			nsname:              egressTestSvcNamespace,
+			sourceIPBy:          "LoadBalancerIP",
+			addToRuntimeObjects: true,
+			client:              true,
+			expectedError:       nil,
+		},
+		{
+			name:                egressTestSvcName,
+			nsname:              egressTestSvcNamespace,
+			sourceIPBy:          "LoadBalancerIP",
+			addToRuntimeObjects: false,
+			client:              true,
+			expectedError:       fmt.Errorf("egressservices.k8s.ovn.org %q not found", egressTestSvcName),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		var svcName string
+
+		if testCase.expectedError != nil {
+			svcName = "hello"
+		} else {
+			svcName = testCase.name
+		}
```

Comment:
I would use the approach of [including the builder in each test case](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/mco/mcp_test.go#L131) so that you don't need to worry about updating the name here or anything

Reply:
updated